underscore.js

```diff
@@ -490,17 +490,20 @@
   };
 
   // Internal implementation of a recursive `flatten` function.
-  var flatten = function(input, shallow, strict) {
-    var output = [], idx = 0;
+  var flatten = function(input, shallow, strict, _output) {
+    var output = _output ? _output : [];
```

Comment:
can you rename the param output and do

output = output || [];

I'll merge this in a bit